cmd/influxd/config_test.go

```diff
@@ -61,12 +62,40 @@ func TestParseConfig(t *testing.T) {
 		t.Fatalf("http api ssl cert path mismatch: %v", c.HTTPAPI.SSLCertPath)
 	}
 
-	if c.InputPlugins.Graphite.Enabled != false {
-		t.Fatalf("graphite enabled mismatch: %v", c.InputPlugins.Graphite.Enabled)
-	} else if c.InputPlugins.Graphite.Port != 2003 {
-		t.Fatalf("graphite port mismatch: %v", c.InputPlugins.Graphite.Enabled)
-	} else if c.InputPlugins.Graphite.Database != "" {
-		t.Fatalf("graphite database mismatch: %v", c.InputPlugins.Graphite.Database)
+	if len(c.Graphite) != 2 {
+		t.Fatalf("graphites  mismatch.  expected %v, got: %v", 2, len(c.Graphite))
+	}
+
+	tcpGraphite := c.Graphite[0]
+	switch {
+	case tcpGraphite.Enabled != true:
+		t.Fatalf("graphite tcp enabled mismatch: expected: %v, got %v", true, tcpGraphite.Enabled)
+	case tcpGraphite.Address != "192.168.0.1":
+		t.Fatalf("graphite tcp address mismatch: expected %v, got  %v", "192.168.0.1", tcpGraphite.Address)
+	case tcpGraphite.Port != 2003:
+		t.Fatalf("graphite tcp port mismatch: expected %v, got %v", 2003, tcpGraphite.Port)
+	case tcpGraphite.Database != "graphite_tcp":
+		t.Fatalf("graphite tcp database mismatch: expected %v, got %v", "graphite_tcp", tcpGraphite.Database)
+	case strings.ToLower(tcpGraphite.Protocol) != "tcp":
+		t.Fatalf("graphite tcp protocol mismatch: expected %v, got %v", "tcp", strings.ToLower(tcpGraphite.Protocol))
+	case tcpGraphite.NamePosition != "last":
+		t.Fatalf("graphite tcp name-position mismatch: expected %v, got %v", "last", tcpGraphite.NamePosition)
+	case tcpGraphite.NameSeparator != "-":
+		t.Fatalf("graphite tcp name-separator mismatch: expected %v, got %v", "-", tcpGraphite.NameSeparator)
+	}
```

Comment:
I tend to use `switch` only when I have a single value I'm switching on. Using conditionals in the cases feels weird to me since that's the syntax of an `if-elseif`: `if <condition> {} else if <condition> {} ...`. Currently the rest of the test suite uses `if-elseif`. If you're not married to a style then I'd prefer it to be consistent.